demo/admin/src/products/ProductForm.tsx

```diff
@@ -56,8 +64,13 @@ const rootBlocks = {
     image: DamImageBlock,
 };
 
-type FormValues = Omit<GQLProductFormManualFragment, "image"> & {
+type FormValues = Omit<GQLProductFormManualFragment, "image" | "manufacturer"> & {
     image: BlockState<typeof rootBlocks.image>;
+    manufacturerFilter?: { id: string };
+    manufacturer?: {
+        id: string;
+        name: string;
+    };
```

Comment:
why can't you use manufacturer from the generated type?

Reply:
because I'm also fetching country to pre-fill the filter-field and also would manufacturer from the generated type not match the form-state value of manufacturer-async-select (only id and name is fetched).

Reply:
you could use a query like that
```
manufacturer {
    id
    name
}
manufacturerCountry: manufacturer {
    country
}
```

Reply:
that's a handy trick :-)

updated the code to use this.